source/llassetgen/source/Image.cpp

```diff
@@ -0,0 +1,367 @@
+    Image::Image(const size_t width, const size_t height, const uint8_t _bitDepth) {
+        assert(_bitDepth == 1 || _bitDepth == 2 || _bitDepth == 4 || _bitDepth == 8 || _bitDepth == 16 || _bitDepth == 24 || _bitDepth == 32);
+        min.x = 0;
+        min.y = 0;
+        max.x = width;
+        max.y = height;
+        bitDepth = _bitDepth;
+        stride = (size_t)std::ceil(float(getWidth() * (float(bitDepth) / 8.0f)));
+        data = std::shared_ptr<uint8_t>(new uint8_t[getHeight() * stride]());
+    }
+
+    size_t Image::getWidth() const {
+        return max.x - min.x;
+    }
+
+    size_t Image::getHeight() const {
+        return max.y - min.y;
+    }
+
+    size_t Image::getBitDepth() const {
+        return bitDepth;
+    }
+
+	Image Image::view(const Vec2<size_t> _min, const Vec2<size_t> _max) {
+        return Image(_min, _max, stride, bitDepth, data);
+    }
+
+    Image::Image(const Vec2<size_t> _min, const Vec2<size_t> _max, const size_t _stride, const uint8_t _bitDepth, const std::shared_ptr<uint8_t>_data) {
+        min = _min;
+        max = _max;
+        stride = _stride;
+        bitDepth = _bitDepth;
+        data = _data;
+    }
+
+    bool Image::isValid(const Vec2<size_t> pos) const {
+        Vec2<size_t> offset = min + pos;
+        return offset.x >= min.x && offset.x < max.x && offset.y >= min.y && offset.y < max.y;
+    }
+
+    template<typename pixelType>
+    pixelType Image::getPixel(const Vec2<size_t> pos) const {
+        assert(isValid(pos));
+        Vec2<size_t> offset = min + pos;
+        if (bitDepth <= 8) {
+            uint8_t byte = data.get()[offset.y * stride + offset.x * bitDepth / 8];
+            size_t bit_pos = offset.x % (8 / bitDepth);
+            byte <<= bit_pos * bitDepth;
+            byte >>= 8 - bitDepth;
+            uint32_t casted = static_cast<uint32_t>(byte);
+            return *reinterpret_cast<pixelType*>(&casted);
+        } else {
+            uint32_t return_data = 0;
+            for (size_t byte_pos = 0; byte_pos < bitDepth / 8; ++byte_pos) {
+                return_data <<= 8;
+                return_data |= data.get()[offset.y * stride + offset.x * bitDepth / 8 + byte_pos];
+            }
+            return *reinterpret_cast<pixelType*>(&return_data);
+        }
+    }
+
+    template LLASSETGEN_API float Image::getPixel<float>(const Vec2<size_t> pos) const;
+    template LLASSETGEN_API uint32_t Image::getPixel<uint32_t>(const Vec2<size_t> pos) const;
+    template LLASSETGEN_API uint16_t Image::getPixel<uint16_t>(const Vec2<size_t> pos) const;
+    template LLASSETGEN_API uint8_t Image::getPixel<uint8_t>(const Vec2<size_t> pos) const;
+
+    template<typename pixelType>
+    void Image::setPixel(const Vec2<size_t> pos, pixelType in) const {
+        assert(isValid(pos));
+        Vec2<size_t> offset = min + pos;
+        if (bitDepth <= 8) {
+            uint8_t mask = 0xFF;
+            uint8_t in_byte = static_cast<uint8_t>(*reinterpret_cast<uint32_t*>(&in));
```

Comment:
In [DistanceTransform.cpp#L55](https://github.com/hpicgs/openll-asset-generator/blob/bitmap_class/source/llassetgen/source/DistanceTransform.cpp#L55) this function is called either with `in` set to 0 or infinity. In both cases `in_byte` would be 0.